### PR TITLE
Add map regions

### PR DESCRIPTION
**Pull request by jonathonlui**
I didn't know if SCALE was going to have an official Guidebook this year so last night I had started scraping the SCALE schedule website to create my own Guidebook. I had implemented the map regions feature in my guidebook.

Today I saw the official Guidebook was published so I thought it'd be interesting to also add map region support to this sync script. I'm assuming SCALE's Guidebook account is a premium account that supports maps.

Three's no public API for map regions so this addition uses an undocumented API used by the Guidebook builder website.

**One  caveat**: when a location is linked to a map region the location's type is changed to `gb-interactive` which then "hides" it from the public API. It'll probably cause some issues when because after adding map regions the `self.rooms` will be empty and instead be in `self.x_rooms`.

How to use:

1. Create new map in Guidebook's web builder using the file `Pasadena-Convention-Center-Map-1000-72-fs8.png`
2. Find the API key the Guidebook's web builder is using. One was is to open the builder website and use this is url bar: 
    ```javascript
    javascript:alert(JSON.parse(localStorage.getItem('CurrentUser')).jwt)
    ```
3. Add the tokento a file called`guidebook_api_x.txt` (if this file doesn't exist the sync script won't try to sync map regions).

Run the sync script. This will update the location types to `gb-interactive` and create the map regions.

### CODE REVIEW

**Comment by jaymzh** [comment]
Thanks for this, that's interesting.

I'm wary of hiding our locations, it means we won't be able to update room info anymore.... and in fact, we'll attempt to add it every time we run it... but I'll play with this when I get a moment.

**Comment by jonathonlui** [comment]
As an alternative uploading the map image to the photo sections works too. Then add a note about it to the guildebook description.

File: guidebook/sync_guidebook.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
 # to update that.
 #
 
+from __future__ import print_function
```
Comment (jaymzh) [inline guidebook/sync_guidebook.py]: this is like the primary reason I wrote it in py2, btw. :)

File: guidebook/sync_guidebook.py [hunk 2]
```diff
@@ -220,6 +285,52 @@ def setup_rooms(self, rooms):
...
+            'map_object': self.x_map_id,
+            'location': location_id,
+            'relative_x': map_region['x'],
+            'relative_y': map_region['y'],
+            'relative_width': map_region['w'],
+            'relative_height': map_region['h'],
+        }
+        self.add_thing('x-map-regions', name, data, update, rid)
+
+    def get_x_map_region_for_room(self, room):
```
Comment (jaymzh) [inline guidebook/sync_guidebook.py]: This method doesn't work. None of these lcations have an ID, they look like this:

```
{u'guide': 125002, u'last_updated': u'2018-03-07T02:54:50.156424+0000', u'uuid': u'0c809b8e-1f86-11e8-b6e5-0ec2a56d2ae2', u'url': u'https://builder.guidebook.com/api/locations/2725731/', u'created_at': u'2018-03-04T08:28:43.067192+0000', u'description': None, u'longitude': None, u'id': 2725731, u'address': None, u'latitude': None, u'location_type': u'gb-interactive', u'name': {u'en-US': u'Ballroom H'}}
```

**Comment by jaymzh** [comment]
ALright I have a version of this that owrks....

**Comment by jaymzh** [comment]
Merged in a8564da9dcee640914df14f7c6e84627fdf627df along with a fixer commit.

**Pull request closed without merging**